wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -245,46 +262,57 @@ type Coins = Int
 
 -- | A type modelling the request for a new wallet.
 data NewWallet = NewWallet {
-      newwalBackupPhrase :: BackupPhrase
-    , newwalPassphrase   :: Maybe Passphrase
+      newwalBackupPhrase     :: !BackupPhrase
+    -- ^ The backup phrase to restore the wallet.
+    , newwalSpendingPassword :: !(Maybe SpendingPassword)
+    -- ^ The spending password to encrypt the private keys.
+    , newwalAssuranceLevel   :: !AssuranceLevel
+    , newwalName             :: !Text
     } deriving (Eq, Show, Generic)
 
 deriveJSON Serokell.defaultOptions  ''NewWallet
 
 instance Arbitrary NewWallet where
-  arbitrary = NewWallet <$> pure "MyBackupPhraseHashed"
+  arbitrary = NewWallet <$> arbitrary
                         <*> pure (Just "My passphrase")
+                        <*> arbitrary
+                        <*> pure "My Wallet"
 
 -- | A type modelling the update of an existing wallet.
 data WalletUpdate = WalletUpdate {
-      uwalUnit      :: !Int
-    , uwalAssurance :: WalletAssurance
-    , uwalName      :: Text
+      uwalAssuranceLevel :: !AssuranceLevel
+    , uwalName           :: !Text
     } deriving (Eq, Show, Generic)
 
 deriveJSON Serokell.defaultOptions  ''WalletUpdate
 
 instance Arbitrary WalletUpdate where
-  arbitrary = WalletUpdate <$> fmap getPositive arbitrary
-                           <*> arbitrary
-                           <*> fmap fromString arbitrary
+  arbitrary = WalletUpdate <$> arbitrary
+                           <*> pure "My Wallet"
+
+-- A 'RenderedBalance' represent the @rendered@ UTXO for
+-- this 'Wallet'.
+newtype RenderedBalance = RenderedBalance { renderBalance :: Text }
+                        deriving (Show, Eq, Generic)
```

Comment:
Comment is incorrect, I believe you wanted to say not "UTXO", but "balance".
Also, as I already said, I'm really against using such proxy types in new wallet. I'm really for simply using `Coin` here (and making a JSON instance which parses `Coin` from text). I don't see any particular reason we should store and manipulate balance as simple `Text` on backend side -- if you do see, please enlighten me.

Reply:
> Comment is incorrect, I believe you wanted to say not "UTXO", but "balance".

I will bite the bullet and look like a noob (which I am 😂 ): isn't - practically speaking - the UTXO for a wallet also equivalent to its balance?  

> Also, as I already said, I'm really against using such proxy types in new wallet. 

I hear you, but I think at the same time there are situation when we cannot avoid them, because we have an _impedance mismatch_ between the client types and the backend types, which is what you guys learnt the hard way with the old wallet (cfr. the proliferation of C-prefixed types, like `CCoin` & Co.

I feel there is a tension in the design space we will need to investigate; from one side we don't want all this extra types, but on the other we cannot overload our `Aeson` instances with too much logic (but maybe I'm wrong and you can convince me otherwise 😀 ). But for sure there are situation when we cannot avoid these client types, for example when creating or updating a wallet (for example). Let's discuss!

Reply:
Well, UTXO is an abbreviation for "Unspent Transaction Outputs", which means that UTXO denotes a set of unspent transaction outputs (surprisingly enough). Unspent transaction output is a pair `(TxIn, TxOut)`, where `TxIn` is basically a pair of transaction hash and index of given output in this transaction, and `TxOut` is a pair of output address and amount of money associated with this output. Balance is simply an unspent amount of money on address, so it's a single `Coin` value. They are not equivalent -- you count balance if you know Utxo, but not the other way round.

Continuing this argument about proxy types:
- Please read my comment after Duncan's comment above
- I'm not sure what do you mean by _impedance mismatch_ here.
- C-prefixed types was introduced in order to generate Purescript types with similar JSON instances automatically via `purescript-bridge`. We couldn't do this for core types, e.g. `Address`, because it made no sense to introduce all the complicated internal structure of actual `Address` datatype to the client. And I think that this was a _bad_ design decision. Frontend guys were pissed off by purescript bridge, and we were suffering because of all this damned C-types. As soon as we're getting rid of purescript bridge, there's no more reason to bear with suffering from proxy types anymore.